website/docs/mutation.md

```diff
@@ -0,0 +1,284 @@
+- conditional - conditions under which the mutation will be applied
+
+#### Scope
+
+The scope section describes the resource which will be mutated.
+It allows to filter the resources to be mutated by kind, label and namespace.
+
+An example of the scope section.
+```yaml
+    applyTo:
+        - groups: [""]
+          kinds: ["Pod"]
+          versions: ["v1"]
+    match:
+        scope: Namespaced | Cluster
+        kinds: 
+        - APIGroups: []
+          kinds: []
+        labelSelector: []
+        namespaces: []
+        namespaceSelector: []
+        excludedNamespaces: []
+```
+
+Note that the `applyTo` section applies to the Assign CRD only. It allows filtering of resources by the resource GVK (group version kind). Note that the `applyTo` section does not accept globs.
+
+The `match` section is common to both Assign and AssignMetadata. It supports the following elements:
+- scope - the scope (Namespaced | Cluster) of the mutated resource
+- kinds - the resource kind, any of the elements listed
+- labelSelector - filters resources by resource labels listed
+- namespaces - list of allowed namespaces, only resources in listed namespaces will be mutated
+- namespaceSelector - filters resources by namespace selector
+- excludedNamespaces - list of excluded namespaces, resources in listed namespaces will not be mutated
+
+Note that the resource is not filtered if an element is not present or an empty list.
+
+#### Intent
+
+This specifies what should be changed in the resource.
+
+An example of the section is shown below:
+```yaml
+    location: "spec.containers[name:foo].imagePullPolicy"
+    parameters:
+       assign:
+        value: "Always"
+```
+
+The `location` element specifies the path to be modified.
+The `parameters.assign.value` element specifies the value to be set for the element specified in `location`. Note that the value can either be a simple string or a composite value.
+
+An example of a composite value:
+```yaml
+    location: "spec.containers[name:networking]"
+    parameters: 
+      assign: 
+        value: 
+          name: "networking"
+          imagePullPolicy: Always
+
+```
+
+The `location` element can specify either a simple subelement or an element in a list.
+For example the location `spec.containers[name:foo].imagePullPolicy` would be parsed as follows:
+- ***spec**.containers[name:foo].imagePullPolicy* - the spec element
+- *spec.**containers[name:foo]**.imagePullPolicy* - container subelement of spec. The container element is a list. Out of the list choosen, an element with the `name` element having the value `foo`.
+ - *spec.containers[name:foo].**imagePullPolicy*** - in the element from the list chosen in the previous step the element `imagePullPolicy` is chosen
+
+The yaml illustrating the above `location`:
+```yaml
+spec:
+   containers:
+   -  name:foo
+      imagePullPolicy:
+```
+Note that when the list element is empty (`spec.containers[].imagePullPolicy`), all the list element will be updated.
```

Comment:
I don't think this is true? I think the key field must be provided:

`spec.containers[name:*].imagePullPolicy`

if not, we should fix that

Reply:
Replaced with:
Wildcards can be used for list element values: `spec.containers[name:*].imagePullPolicy`
I did not check this, but I thought that this is the desired behavior. Let me check that later.

Reply:
We need the key field to make sure users aren't overwriting key values.

I wonder if we are denying requests like:

```
spec.containers[name:*] = {name: "foo", ...}
```

if not, we should be, as wildcards only really make sense when we're setting fields of list members.